src/software/thunderscope/binary_context_managers.py

```diff
@@ -528,19 +540,283 @@ def send_ci_input(
             ci_input.change.CopyFrom(ci_change)
             ci_ci_input.api_inputs.append(ci_input)
 
-        # https://cwiki.apache.org/confluence/display/GEODE/Delimiting+Protobuf+Messages
-        size = ci_ci_input.ByteSize()
+        ci_output_list = self.send_ci_input(ci_ci_input)
+
+        return ci_output_list
+
+    def send_ci_input(self, ci_input):
+        """
+        Send CiInput proto to the Gamecontroller. Retries if the Gamecontroller output isn't parseable as a CiOutput proto
+
+        :param CiInput proto to send to the Gamecontroller
+
+        :return: a list of CiOutput protos received from the Gamecontroller
+        """
+        ci_output_list = list()
+        while True:
+            try:
+                self.ci_socket.send(ci_input)
+                ci_output_list = self.ci_socket.receive(CiOutput)
+                break
+            except SslSocketProtoParseException as parse_err:
+                logging.info(
+                    "error receiving CiOutput proto from the gamecontroller: "
+                    + parse_err.args
+                )
+
+        return ci_output_list
+
+    def reset_team(self, name, team):
+        """
+        Returns an UpdateTeamState proto for the gamecontroller to reset team info.
+
+        :param name name of the new team
+        :param team yellow or blue team to update
+
+        :return: corresponding UpdateTeamState proto
+        """
+        update_team_state = UpdateTeamState()
+        update_team_state.for_team = team
+        update_team_state.team_name = name
+        update_team_state.goals = 0
+        update_team_state.timeouts_left = 4
+        update_team_state.timeout_time_left = "05:00"
+        update_team_state.can_place_ball = True
+
+        return update_team_state
+
+    def reset_game(self, division):
+        """
+        Returns an UpdateConfig proto for the Gamecontroller to reset game info.
+
+        :param division the Division proto corresponding to the game division to set up the Gamecontroller for
+
+        :return: corresponding UpdateConfig proto
+        """
+        game_update = UpdateConfig()
+        game_update.division = division
+        game_update.first_kickoff_team = Team.BLUE
+        game_update.auto_continue = True
+        game_update.match_type = MatchType.FRIENDLY
+
+        return game_update
+
+    def reset_team_info(self, division):
+        """
+        Sends a message to the Gamecontroller to reset Team information.
+
+        :param division the Division proto corresponding to the game division to set up the Gamecontroller for
+
+        :return: a list of CiOutput protos from the Gamecontroller
+        """
+        ci_ci_input = CiInput(timestamp=int(time.time_ns()))
```

Comment:
Could you add some comments and/or possibly update the variable names here? I'm finding it hard to follow what ci_ci_... vs ci_... mean

Reply:
updated naming